### DISCUSSION TITLE
Claims

### DISCUSSION THREAD

**Discussion by Teddy293** [GriefPrevention/GriefPrevention#2398] [Ideas]
**As** a player/owner of a server
**I want** to be able to be able to do like a command /claimnear 
**Because** I would like to see if there are any nearby claim or see how far the nearest claim is 

**As** a player/owner of a server
**I want** to be able to have a edit where players can allow players to kill things in there claim for certain animals/mobs
**Because** like mob farms for example zombie spawners spawn chickens allow players to kill chickens in that claim etc..

**Comment by Jikoo** [comment]
If you have multiple requests it's easier for us if you post them separately. It also makes it easier to track whether or not something is resolved.

> As a player/owner of a server
I want to be able to be able to do like a command /claimnear
Because I would like to see if there are any nearby claim or see how far the nearest claim is

You can use sneak+right click with the inspection tool to visualize nearby claims. Permission can be granted to regular players with `griefprevention.visualizenearbyclaims`. https://legacy.griefprevention.com/permissions/
A distance-to-nearest command would be pretty inefficient with GP's current model. An addon could do it relatively easily by maintaining copies of claim volumes in an appropriate bounding volume hierarchy (If memory serves, a K-D tree would probably be good).

> As a player/owner of a server
I want to be able to have a edit where players can allow players to kill things in there claim for certain animals/mobs
Because like mob farms for example zombie spawners spawn chickens allow players to kill chickens in that claim etc..

`/containertrust` allows killing farm animals. You can use a subclaim to reduce the amount of area you trust people in.
